pinot-core/src/main/java/org/apache/pinot/core/query/aggregation/function/FrequentLongsSketchAggregationFunction.java

```diff
@@ -0,0 +1,251 @@
+
+
+/**
+ * <p>
+ *  {@code FrequentLongsSketchAggregationFunction} provides an approximate FrequentItems aggregation function based on
+ *  <a href="https://datasketches.apache.org/docs/Frequency/FrequentItemsOverview.html">Apache DataSketches library</a>.
+ *  It is memory efficient compared to exact counting.
+ * </p>
+ * <p>
+ *   The function takes an INT or LONG column as input and returns a Base64 encoded sketch object which can be
+ *   deserialized and used to estimate the frequency of items in the dataset (how many times they appear).
+ * </p>
+ * <p><b>FREQUENT_STRINGS_SKETCH(col, maxMapSize=256)</b></p>
+ * <p>E.g.:</p>
+ * <ul>
+ *   <li><b>FREQUENT_LONGS_SKETCH(col)</b></li>
+ *   <li><b>FREQUENT_LONGS_SKETCH(col, 1024)</b></li>
+ * </ul>
+ *
+ * <p>
+ *   If the column type is BYTES, the aggregation function will assume it is a serialized FrequentItems data sketch
+ *   of type `LongsSketch`and will attempt to deserialize it for merging with other sketch objects.
+ * </p>
+ *
+ * <p>
+ *   There is a variation of the function (<b>FREQUENT_STRINGS_SKETCH</b>) which accepts STRING type input columns.
+ * </p>
+ */
+public class FrequentLongsSketchAggregationFunction
+    extends BaseSingleInputAggregationFunction<LongsSketch, Comparable<?>> {
+  protected static final int DEFAULT_MAX_MAP_SIZE = 256;
+
+  protected int _maxMapSize;
+
+  public FrequentLongsSketchAggregationFunction(List<ExpressionContext> arguments) {
+    super(arguments.get(0));
+    int numArguments = arguments.size();
+    Preconditions.checkArgument(numArguments == 1 || numArguments == 2,
+        "Expecting 1 or 2 arguments for FrequentLongsSketch function: FREQUENTITEMSSKETCH(column, maxMapSize");
+    _maxMapSize = numArguments == 2 ? arguments.get(1).getLiteral().getIntValue() : DEFAULT_MAX_MAP_SIZE;
+  }
+
+  @Override
+  public AggregationFunctionType getType() {
+    return AggregationFunctionType.FREQUENTLONGSSKETCH;
+  }
+
+  @Override
+  public AggregationResultHolder createAggregationResultHolder() {
+    return new ObjectAggregationResultHolder();
+  }
+
+  @Override
+  public GroupByResultHolder createGroupByResultHolder(int initialCapacity, int maxCapacity) {
+    return new ObjectGroupByResultHolder(initialCapacity, maxCapacity);
+  }
+
+  @Override
+  public void aggregate(int length, AggregationResultHolder aggregationResultHolder,
+      Map<ExpressionContext, BlockValSet> blockValSetMap) {
+    BlockValSet valueSet = blockValSetMap.get(_expression);
+    FieldSpec.DataType valueType = valueSet.getValueType();
+
+    LongsSketch sketch = getOrCreateSketch(aggregationResultHolder);
+
+    if (valueType == FieldSpec.DataType.BYTES) {
+      // Assuming the column contains serialized data sketch
+      LongsSketch[] deserializedSketches =
+          deserializeSketches(blockValSetMap.get(_expression).getBytesValuesSV());
+      sketch = getOrCreateSketch(aggregationResultHolder);
+
+      for (int i = 0; i < length; i++) {
```

Comment:
NIT: can we use foreach syntax to make it more readable?

Reply:
Done